examples/telco/notebook.jl

```diff
@@ -0,0 +1,756 @@
+
+predict(mach, rows=71:73)
+
+# or on completely new data, as in
+
+Xnew = (sepal_length = [5.1, 6.3],
+        sepal_width = [3.0, 2.5],
+        petal_length = [1.4, 4.9],
+        petal_width = [0.3, 1.5])
+yhat = predict(mach, Xnew)
+
+# These are probabilistic predictions which can be manipulated using a
+# widely adopted interface defined in the Distributions.jl
+# package. For example, we can get raw probabilities like this:
+
+pdf.(yhat, "virginica")
+
+
+# We now turn to the Telco dataset.
+
+
+# ## Getting the Telco data
+
+import DataFrames
+
+#-
+
+data = OpenML.load(42178) # data set from OpenML.org
+df0 = DataFrames.DataFrame(data)
+first(df0, 4)
+
+# The object of this tutorial is to build and evaluate supervised
+# learning models to predict the `:Churn` variable, a binary variable
+# measuring customer retention, based on other variables that are
+# relevant.
+
+# In the table, observations correspond to rows, and features to
+# columns, which is the convention for representing all
+# two-dimensional data in MLJ.
+
+
+# ## Type coercion
+
+# > Introduces: `scitype`, `schema`, `coerce`
+
+# A ["scientific
+# type"](https://juliaai.github.io/ScientificTypes.jl/dev/) or
+# *scitype* indicates how MLJ will *interpret* data. For example,
+# `typeof(3.14) == Float64`, while `scitype(3.14) == Continuous` and
+# also `scitype(3.14f0) == Continuous`. In MLJ, model data
+# requirements are articulated using scitypes.
+
+# Here are common "scalar" scitypes:
+
+# ![](assets/scitypes.png)
+
+# There are also container scitypes. For example, the scitype of any
+# `N`-dimensional array is `AbstractArray{S, N}`, where `S` is the scitype of the
+# elements:
+
+scitype(["cat", "mouse", "dog"])
+
+# The `schema` operator summarizes the column scitypes of a table:
+
+schema(df0) |> DataFrames.DataFrame  # converted to DataFrame for better display
+
+# All of the fields being interpreted as `Textual` are really
+# something else, either `Multiclass` or, in the case of
+# `:TotalCharges`, `Continuous`. In fact, `:TotalCharges` is
+# mostly floats wrapped as strings. However, it needs special
+# treatment because some elements consist of a single space, " ",
+# which we'll treat as "0.0".
+
+fix_blanks(v) = map(v) do x
+    if x == " "
+        return "0.0"
+    else
+        return x
+    end
+end
+
+df0.TotalCharges = fix_blanks(df0.TotalCharges);
+
+# Coercing the `:TotalCharges` type to ensure a `Continuous` scitype:
+
+coerce!(df0, :TotalCharges => Continuous);
+
+# Coercing all remaining `Textual` data to `Multiclass`:
+
+coerce!(df0, Textual => Multiclass);
+
+# Finally, we'll coerce our target variable `:Churn` to be
+# `OrderedFactor`, rather than `Multiclass`, to enable a reliable
+# interpretation of metrics like "true positive rate".  By convention,
+# the first class is the negative one:
+
+coerce!(df0, :Churn => OrderedFactor)
+levels(df0.Churn) # to check order
+
+# Re-inspecting the scitypes:
```

Comment:
it would be also good to see `df0` at this point to see what is the physical representation of the data after coercion.